src/diskio/mod.rs

```
@@ -196,18 +196,19 @@ pub fn get_executor<'a>(
     notify_handler: Option<&'a dyn Fn(Notification<'_>)>,
 ) -> Box<dyn Executor + 'a> {
     // If this gets lots of use, consider exposing via the config file.
-    if let Ok(thread_str) = process().var("RUSTUP_IO_THREADS") {
-        if thread_str == "disabled" {
-            Box::new(immediate::ImmediateUnpacker::new())
-        } else if let Ok(thread_count) = thread_str.parse::<usize>() {
-            Box::new(threaded::Threaded::new_with_threads(
-                notify_handler,
-                thread_count,
-            ))
-        } else {
-            Box::new(threaded::Threaded::new(notify_handler))
+    let thread_count = match process()
+        .var("RUSTUP_IO_THREADS")
+        .map(|x| x.parse::<usize>())
+    {
+        Ok(Ok(n)) => n,
+        Ok(Err(_)) => {
+            eprintln!("Invalid value for RUSTUP_IO_THREADS. Has to be an integer");
```

Please don't use eprintln! - either use chain_err or a notify callback message. Also it doesn't have to be an integer - it has to be a natural number.

---

If I use `chain_err` that means you want me to return an error instead of falling back to `num_cpus::get()`? Sure. I can make the method fallible. Going to be some more work, but if you prefer that.

Yeah I know about the natural numbers. I just thought this sounded better even though it's not technically correct. People must understand that negative numbers are not OK :D But I'll change it.

---

Done